src/arkode/arkode_io.c

```diff
@@ -34,6 +35,347 @@
+                           ARKodeSetAdaptivityAdjustment,
+                           ARKodeSetSmallNumEFails,
+                           ARKodeSetMaxErrTestFails,
+                           ARKodeSetMaxConvFails};
+
+  int num_long_keys = 1;
+  const char* long_keys[] = {"arkode.max_num_steps"};
+  arkLongSetFn long_set[] = {ARKodeSetMaxNumSteps};
+
+  int num_real_keys = 18;
+  const char* real_keys[] = {"arkode.nonlin_crdown",
+                             "arkode.nonlin_rdiv",
+                             "arkode.delta_gamma_max",
+                             "arkode.nonlin_conv_coef",
+                             "arkode.init_step",
+                             "arkode.min_step",
+                             "arkode.max_step",
+                             "arkode.stop_time",
+                             "arkode.fixed_step",
+                             "arkode.step_direction",
+                             "arkode.cfl_fraction",
+                             "arkode.safety_factor",
+                             "arkode.error_bias",
+                             "arkode.max_growth",
+                             "arkode.min_reduction",
+                             "arkode.max_first_growth",
+                             "arkode.max_efail_growth",
+                             "arkode.max_cfail_growth"};
+  arkRealSetFn real_set[] = {ARKodeSetNonlinCRDown,
+                             ARKodeSetNonlinRDiv,
+                             ARKodeSetDeltaGammaMax,
+                             ARKodeSetNonlinConvCoef,
+                             ARKodeSetInitStep,
+                             ARKodeSetMinStep,
+                             ARKodeSetMaxStep,
+                             ARKodeSetStopTime,
+                             ARKodeSetFixedStep,
+                             ARKodeSetStepDirection,
+                             ARKodeSetCFLFraction,
+                             ARKodeSetSafetyFactor,
+                             ARKodeSetErrorBias,
+                             ARKodeSetMaxGrowth,
+                             ARKodeSetMinReduction,
+                             ARKodeSetMaxFirstGrowth,
+                             ARKodeSetMaxEFailGrowth,
+                             ARKodeSetMaxCFailGrowth};
+
+  int num_bool_keys = 4;
+  const char* bool_keys[] = {"arkode.nonlinear",
+                             "arkode.clear_stop_time",
+                             "arkode.no_inactive_root_warn",
+                             "arkode.reset_accumulated_error"};
+  arkBoolSetFn bool_set[] = {ARKodeSetNonlinear,
+                             ARKodeClearStopTime,
+                             ARKodeSetNoInactiveRootWarn,
+                             ARKodeResetAccumulatedError};
+
+  int i, j, retval;
+  for (i = 1; i < argc; i++)
+  {
+    sunbooleantype arg_used = SUNFALSE;
+
+    /* skip command-line arguments that do not begin with "arkode." */
+    if (strncmp(argv[i], "arkode.", 7) != 0) { continue; }
```

Comment:
Since we filter by the `arkode.` prefix here, I think it's redundant to include it in all the array entries above. We would just need to shift the string pointer below.

Reply:
Possibly out of scope for this initial PR, but in order to support full configuration over an MRI or operator splitting method where they may be several arkode instances, we may need to support user-specified prefixes. For example,  `arkode.order 2 partition1.order 2 partition2.linear 1`

Reply:
I think we could do this by requiring a certain format for the keys e.g., `user_label.arkode.option` or `arkode.user_label.option`. Then `ARKodeSetFromCommandLine` could accept an additional input for which `user_label` to filter for and if none is provided, only read `arkode.` keys.

Reply:
We have a rough sketch of scheme for MRI and operator splitting in the design document.

Reply:
Ah, I remember this now. Looks like we settled on a `pkg.scope.option` style.

If this functions is going to be at the  ARKODE level, I think it will need an additional input to filter with since this function won't know which MRI level or splitting partition the memory corresponds to. 

Will we also need stepper specific versions of this function to handle options like setting the method to use or should we have only stepper-specific versions that call this function internally?